src/NewTwitchApi/Resources/EventSubApi.php

```diff
@@ -0,0 +1,286 @@
+        }
+
+        $this->subscribe(
+            sprintf('channel.hype_train.%s', $event),
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelmoderatoradd
+     */
+    public function subscribeToChannelModeratorAdd(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelModerator($twitchId, 'add', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelmoderatorremove
+     */
+    public function subscribeToChannelModeratorRemove(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelModerator($twitchId, 'remove', $callback, $bearer, $secret);
+    }
+
+    private function subscribeToChannelHypeTrain(string $twitchId, string $event, string $callback, string $bearer, string $secret): void
+    {
+        if (!in_array($event, ['begin', 'progress', 'end'])) {
+            throw new \InvalidArgumentException('Invalid value for channel_hype_train event type. Accepted values: begin,progress,end.');
+        }
+
+        $this->subscribe(
+            sprintf('channel.hype_train.%s', $event),
+            '1',
+            ['broadcaster_user_id' => $twitchId],
+            $callback,
+            $bearer,
+            $secret
+        );
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelhype_trainbegin
+     */
+    public function subscribeToChannelHypeTrainBegin(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'begin', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelhype_trainprogress
+     */
+    public function subscribeToChannelHypeTrainProgress(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'progress', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#channelhype_trainend
+     */
+    public function subscribeToChannelHypeTrainEnd(string $twitchId, string $callback, string $bearer, string $secret): void
+    {
+        $this->subscribeToChannelHypeTrain($twitchId, 'end', $callback, $bearer, $secret);
+    }
+
+    /**
+     * @link https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types#stream-subscriptions
+     */
+    private function subscribeToStream(string $twitchId, string $event, string $callback, string $bearer, string $secret): void
```

Comment:
- Bearer should always be the first parameter
- Since Secrets are required for all Subscriptions, put this directly after the Bearer
- Since Callbacks are required for all Subscriptions, put this after the Secret
- `event` should be renamed to `event_type` so the language is consistent
- Include a comment to define the expected type, instead of the argument exception as noted below, if it goes outdated, that's okay